src/screens/Collectible/Collectible.js

```diff
@@ -166,90 +166,96 @@ class CollectibleScreen extends React.Component<Props> {
         />
       </SlideModal>
     ));
-  }
-
-  render() {
-    const {
-      navigation,
-      collectibles,
-      openSeaTxHistory,
-      history,
-      accounts,
-      theme,
-    } = this.props;
-    const { assetData }: CollectibleNavigationParams = navigation.state.params;
-    const {
+  };
+
+  const isOwned = useMemo(
+    () => collectibles.some((collectible) => collectible.id === id),
+    [collectibles, id],
+  );
+
+  const mappedCollectiblesTransactions = useMemo(
+    () => {
+      const collectiblesTransactions = accountCollectibleHistory[chain] ?? [];
+      return mapTransactionsHistory(
+        collectiblesTransactions,
+        accounts,
+        COLLECTIBLE_TRANSACTION,
+      );
+    },
+    [accountCollectibleHistory, accounts, chain],
+  );
+
+  const transactions = useMemo(
+    () => {
+      const relatedTransactions = mappedCollectiblesTransactions.filter(({
+        assetData: transactionAssetData,
+      }) => transactionAssetData
+        && transactionAssetData.id
+        && transactionAssetData.contractAddress
+        && transactionAssetData.id === id
+        && addressesEqual(transactionAssetData.contractAddress, contractAddress));
```

Comment:
Looks like a good piece of code to exact as `isTransactionMatchingAssetData` + perhaps also `filterTransactionsMatchingAssetData`

Reply:
With `id` here this is linked to Collectibles since ID alone can match multiple collectibles, but what makes them different is contract address. Added as `isMatchingCollectible` util.